src/Database/Adapter/MariaDB.php

```diff
@@ -1303,260 +1303,257 @@ public function updateDocument(string $collection, Document $document): Document
     }
 
     /**
-     * Update Documents in batches
+     * Batch update documents
      *
      * @param string $collection
+     * @param Document $update
      * @param array<Document> $documents
-     * @param int $batchSize
      *
-     * @return array<Document>
+     * @return int
      *
-     * @throws DuplicateException
-     * @throws \Throwable
+     * @throws DatabaseException
      */
-    public function updateDocuments(string $collection, array $documents, int $batchSize = Database::INSERT_BATCH_SIZE): array
+    public function updateDocuments(string $collection, Document $update, array $documents): int
     {
-        if (empty($documents)) {
-            return $documents;
+        $attributes = $update->getAttributes();
+        if (!empty($update->getPermissions())) {
+            $attributes['_permissions'] = json_encode($update->getPermissions());
         }
 
-        try {
-            $name = $this->filter($collection);
-            $batches = \array_chunk($documents, max(1, $batchSize));
+        if (empty($attributes)) {
+            return 0;
+        }
 
-            foreach ($batches as $batch) {
-                $bindIndex = 0;
-                $batchKeys = [];
-                $bindValues = [];
+        $name = $this->filter($collection);
 
-                $removeQuery = '';
-                $removeBindValues = [];
+        $columns = '';
 
-                $addQuery = '';
-                $addBindValues = [];
-                /* @var $document Document */
-                foreach ($batch as $index => $document) {
-                    $attributes = $document->getAttributes();
-                    $attributes['_uid'] = $document->getId();
-                    $attributes['_createdAt'] = $document->getCreatedAt();
-                    $attributes['_updatedAt'] = $document->getUpdatedAt();
-                    $attributes['_permissions'] = json_encode($document->getPermissions());
+        $where = [];
 
-                    if ($this->sharedTables) {
-                        $attributes['_tenant'] = $this->tenant;
-                    }
+        $ids = \array_map(fn ($document) => $document->getId(), $documents);
+        $where[] = "_uid IN (" . \implode(', ', \array_map(fn ($id) => ":_id_{$id}", \array_keys($ids))) . ")";
```

Comment:
Let's give this a better name for clarity:

```suggestion
$where[] = "_uid IN (" . \implode(', ', \array_map(fn ($index) => ":_id_{$index}", \array_keys($ids))) . ")";
```